### PR TITLE
Change structural types scheme

### PR DESCRIPTION
**Pull request by odersky**
Make Selectable a marker trait without any members.

### CODE REVIEW

**Comment by odersky** [comment]
I reverted the previous attempt to use `import language.reflectiveCalls` as an alternative way to import the `reflectiveSelectable` conversion. It caused ambiguities. I briefly considered to make 
`language.reflectiveCalls` the only way to produce a `reflective.Selectable`, but decided against it. Doing this would be a useful hack in the short run, but would destroy all invariants we have for the language` module: Namely, that it is compile-time only and that every entry is mirrored by a command-line option.

File: docs/docs/reference/changed-features/structural-types.md [hunk 1]
```diff
@@ -29,25 +31,91 @@ configure how fields and methods should be resolved.
...
+program would print  "Emma is 42 years old.":
+```scala
+  val person = Record("name" -> "Emma", "age" -> 42).asInstanceOf[Person]
+  println(s"${person.name} is ${person.age} years old.")
+```
+The parent type `Record` in this example is a generic class that can represent arbitrary records in its `elems` argument. This argument is a
+sequence of pairs of labels of type `String` and values of type `Any`.
+When we create a `Person` as a `Record` we have to assert with a typecast
+that the record defines the right fields of the right types. `Record`
+itself is too weakly typed so the compiler cannot know this this without
```
Comment (sjrd) [inline docs/docs/reference/changed-features/structural-types.md:55]:
```suggestion
itself is too weakly typed so the compiler cannot know this without
```

File: docs/docs/reference/changed-features/structural-types.md [hunk 2]
```diff
@@ -29,25 +31,91 @@ configure how fields and methods should be resolved.
...
+help from the user. In practice, the connection between a structural type
+and its underlying generic representation would most likely be done by
+a database layer, and therefore would not be a concern of the end user
```
Comment (sjrd) [inline docs/docs/reference/changed-features/structural-types.md:58]:
```suggestion
a database layer, and therefore would not be a concern of the end user.
```

File: docs/docs/reference/changed-features/structural-types-spec.md [hunk 3]
```diff
@@ -14,61 +14,63 @@ RefineStat    ::= ‘val’ VarDcl | ‘def’ DefDcl | ‘type’ {nl} TypeDcl
...
-takes a method name, `ClassTag`s representing its parameters types and
-the arguments to pass to the function. It will return the result of
-calling this function with the given arguments.
+Implementations of `Selectable` have to make available one or both of
+the methods `selectDynamic` and `applyDynamic`. The methods could be members of the `Selectable` implementation or they could be extension methods.
+
+The `selectDynamic` method takes a field name and returns the value associated with that name in the `Selectable`.
+It should have a signature of the form:
+```scala
+def selectDynamic(name: String): T
```
[the remark this answers is not in the document]
Comment (sjrd) [reply docs/docs/reference/changed-features/structural-types-spec.md:34]: And it would be really good to actually test that. ;)

**Pull request merged**